tutorials/dnssec-cloud-dns-domains/index.md

```diff
@@ -0,0 +1,1162 @@
+possible to activate DNSSEC for a domain in a TLD that does not support it.
+Confirm whether the TLD for your domain supports DNSSEC by looking for it in the
+[ICANN TLD DNSSEC Report][7].
+
+[7]: http://stats.research.icann.org/dns/tld_report/
+
+-   If your TLD answers YES to the first two questions (Signed? and DS in Root?)
+    it supports DNSSEC (these rows are **green**).
+
+-   Most TLDs that do not support DNSSEC (as of October 2017) are two-letter
+    country-code domains (ccTLDs) such as .AI or .AL, but there are also a few
+    global TLDs like .AERO that do not support it (these rows are light gray).
+
+-   Some TLDs have enabled DNSSEC signing, but have not published a DS record in
+    the root zone (their rows are blue in the report). In theory you could add a
+    DS record to such a TLD and have DNSSEC activate automatically when they
+    eventually publish a DS record, but it is unlikely that any domain registrar
+    would support that, and it is not recommended, since you should be aware
+    (and in control) of any change to the DNSSEC status of your domains.
+
+If you cannot reach the ICANN TLD DNSSEC site for any reason, you can check with
+the [Verisign DNSSEC debugger][8]: just enter the TLD name and hit return; if
+you see any red ⊗ errors the TLD does not support DNSSEC (e.g.
+http://dnssec-debugger.verisignlabs.com/aq).
+
+[8]: http://dnssec-debugger.verisignlabs.com/us
+
+#### Domain registrar DNSSEC support
+
+Some domain registrars do not support DNSSEC, or only support DNSSEC for domains
+served from their own name servers. If you can’t find [specific instructions][9]
+below for the domain registrar or reseller, check the [ICANN list][10] of
+DNSSEC-capable domain registrars. Domain registrars may only support DNSSEC for
+some top-level domains (TLDs).
+
+[9]: #domain-registrar-specific-instructions
+[10]: https://www.icann.org/resources/pages/deployment-2012-02-25-en
+
+If DNSSEC is supported by the TLD of your domain, but your domain registrar does
+not support DNSSEC (at all, or for that TLD), you may want to consider
+transferring your domain to another domain registrar that does support DNSSEC
+for that TLD.
+
+Domain registrars may also support only some DNSSEC algorithms; while the
+default settings for DNSSEC in Google Cloud DNS are widely supported, if you
+sign domains with newer algorithms like ECDSA, your domain registrar may not
+support adding a DS record with that algorithm.
+
+### Confirm DNSSEC record propagation
+
+Before activating DNSSEC through your domain registrar, make sure that your zone
+is being served with all DNSSEC data. It can take a few minutes for Google Cloud
+DNS to generate DNSSEC keys and signatures, and even longer for validating
+resolvers to see those updates. If you activate DNSSEC at your domain registrar
+before that is complete, validating resolvers may fail to resolve names for your
+domain until it is fully propagated.
+
```

Comment:
These are very good suggestions, do you think it would be better to do this with more comments in the embedded code or in the explanatory text in the tutorial? I'm inclined to the latter (explanatory text) as there are domain administrators who only use Windows systems.

Google Cloud does provide a free Linux shell environment ([Google Cloud Shell](https://cloud.google.com/shell/)) where you can run these commands (and I will add text that explicitly calls that out) but a Windows-only domain administrator is less likely to read the script than to just run it.

As for checking parent/child NS record consistency, the script uses only the NS records provided by the TLD parent; the exhortation is not to check for parent/child consistency, but to make sure that the Google Cloud DNS name servers are the (only) authoritative servers (since no others will possess the same private DNSSEC keys). I'll see if fixing the script to check NS records returned by the NS listed in the referral response is feasible.

Reply:
To be honest, I would explain both in the tutorial as well as in the code.  I don't think there's that much more explaining to do (maybe a few sentences), but sufficient background can help prevent a user from shooting themselves in the foot by skipping a few steps because they think they got it.